### DISCUSSION TITLE
I'm using protobufjs and I generate static js code. Can I use this if I move it to public/

### DISCUSSION THREAD

**Discussion by hidesminimally** [vercel/next.js#23822] [Help]
As the title says I am using a library that generates static js code and I wonder if I can use this in if I leave it in the public/ directory. When I try this, code that uses that static js unrecognizes the path: 

```
import * as proto from 'generated/compiled';
```

Alternatively, I could generate .JSON code. Same question applies. 

Thanks in advance.

**Comment by akellbl4** [comment]
Hi,
I worked with protobuf and did the same. In my case, I just import that code right in a JS file and used that functions for decoding/encoding the messages.

Is the generated code placed in that folder?

**Reply by hidesminimally** [reply]
Hi, thanks for taking a look. 

The generated code is placed in the public/ directory. Then I try to use it using the code snippet above. But typescript complains that it cannot find that module 
```
src/pages/index.tsx:11:24 - error TS2307: Cannot find module 'generated/compiled' or its corresponding type declarations.

11 import * as proto from 'generated/compiled';
                          ~~~~~~~~~~~~~~~~~~~~


Found 1 error.

error Command failed with exit code 2.
```

**Reply by akellbl4** [reply]
I can suggest placing it somewhere with your js libs/utils or so and import it as you import any other js file.
When you use `'generated/compiled'` webpack trying to resolve node module with the name `generated`

**Reply by hidesminimally** [reply]
Thanks for the suggestion. Maybe a silly question (new to client side work) but wondering if there's any huge benefit for having this static js code in `public/` so it can be served as a static asset from next.js.  I would assume not, but I wanted to check and ask anyway

**Reply by akellbl4** [reply]
everything you put to `public/` will be available via HTTP. For example, you have `public/favicon.ico` and it will be accessible by `http://localhost:3000/favicon.ico`. That means you can put any resources there and then have access to it by url.
